src/model/resources/fusion/fusion.jl

```diff
@@ -0,0 +1,545 @@
+#################################################################
+# Names for keys in an inputs (or future outputs) dictionary
+#################################################################
+
+const FUSION_PULSE_START = "FusionPulseStartVariables"
+const FUSION_PULSE_UNDERWAY = "FusionPulseUnderwayVariables"
+
+const FUSION_PARASITIC_POWER = "FusionParasiticPowerExpressions"
+const FUSION_PULSE_START_POWER = "FusionPulseStartPowerExpressions"
+
+const FUSION_COMPONENT_ZONE = "FusionComponentZones"
+
+#####################################################################
+# Individual fusion reactor component variable and expressions names
+#####################################################################
+
+function fusion_pulse_start_name(resource_component::AbstractString)::String
+    "vFusionPulseStart_" * resource_component
+end
+
+function fusion_pulse_underway_name(resource_component::AbstractString)::String
+    "vFusionPulseUnderway_" * resource_component
+end
+
+function fusion_parasitic_active_name(resource_component::AbstractString)::String
+    "eFusionParasiticActive_" * resource_component
+end
+
+function fusion_parasitic_passive_name(resource_component::AbstractString)::String
+    "eFusionParasiticPassive_" * resource_component
+end
+
+function fusion_parasitic_start_energy_name(resource_component::AbstractString)::String
+    "eFusionParasiticStartEnergy_" * resource_component
+end
+
+function fusion_pulse_start_power_name(resource_component::AbstractString)::String
+    "eFusionParasiticPulseStartPower_" * resource_component
+end
+
+function fusion_parasitic_total_name(resource_component::AbstractString)::String
+    "eFusionParasiticTotal_" * resource_component
+end
+
+#############################################################################
+# Set and get these variables, expressions from the inputs (or outputs) dict
+#############################################################################
+
+@doc raw"""
+    ensure_fusion_expression_records!(dict::Dict)
+
+    dict: a dictionary of model data
+
+    This should be called by each method that adds fusion formulations,
+    to ensure that certain entries in the model data dict exist.
+"""
+function ensure_fusion_expression_records!(dict::Dict)
+    for var in (FUSION_PARASITIC_POWER, FUSION_PULSE_START_POWER)
+        if var ∉ keys(dict)
+            dict[var] = Set{Symbol}()
+        end
+    end
+    var = FUSION_COMPONENT_ZONE
+    if var ∉ keys(dict)
+        dict[var] = Dict{Int, Set{AbstractString}}()
+    end
+end
```

Comment:
This doesn't work because when you call 

`ensure_records!(dict, [FUSION_PULSE_START, FUSION_PULSE_UNDERWAY], Set{Symbol}())`

`Set{Symbol}()` gets evaluated only once, not once per key.

Maybe I could solve this by calling
`ensure_records!( ..., ..., Set{Symbol})` where the function has
`dict[key] = default_value()`
?

What do you think about that?

Reply:
Sorry, there’s an error, `keys_to_check` should be a `Vector{String}`. I’m not sure I fully understood your comment, isn’t this the expected behavior?
```
julia> function ensure_records!(dict::Dict, keys_to_check::Vector{String}, default_value)
           for key in keys_to_check
               if key ∉ keys(dict)
                   dict[key] = default_value
               end
           end
       end
ensure_records! (generic function with 2 methods)

julia> d = Dict()
Dict{Any, Any}()

julia> ensure_fusion_pulse_variable_records!(d)

julia> d
Dict{Any, Any} with 2 entries:
  "FusionPulseUnderwayVariables" => Set{Symbol}()
  "FusionPulseStartVariables"    => Set{Symbol}()

julia> ensure_fusion_expression_records!(d)

julia> d
Dict{Any, Any} with 5 entries:
  "FusionComponentZones"             => Dict{Int64, Set{AbstractString}}()
  "FusionParasiticPowerExpressions"  => Set{Symbol}()
  "FusionPulseUnderwayVariables"     => Set{Symbol}()
  "FusionPulseStartPowerExpressions" => Set{Symbol}()
  "FusionPulseStartVariables"        => Set{Symbol}()
```

Reply:
The problem is that there's only one `Set` object shared by multiple keys. So if I add an element to the `Set` under `d["FusionParasiticPowerExpressions"]` it also appears under `d["FusionPulseStartPowerExpressions"]`.